app/src/main/java/ciserver/PipelineCompiler.java

```diff
@@ -8,44 +8,50 @@
 
 import org.apache.commons.io.IOUtils;
 
+/**
+ * this class is used for the compile stage of our CI pipeline
+ */
 public class PipelineCompiler implements StageTask {
-	public final ArrayList<String> commands = new ArrayList<>();
-
-	PipelineCompiler(String... commands) {
-		this.commands.addAll(Arrays.asList(commands));
-	}
-
-	/**
-	 * Run the specified `commands` in the folder of the pulled repository.
-	 *
-	 * Note: the PipelinePull component or a similar action must be run before this.
-	 *
-	 * @return the status of the compilation action
-	 */
-	public PipelineStatus execute(String pipelineDir, PushEvent event) {
-		var path = String.format("%s/repositories/%s", pipelineDir, event.headCommit.id);
-		var directory = new File(path);
-		directory.mkdirs(); // Make the directories recursively
-
-		try {
-			var process = new ProcessBuilder(commands)
-					.directory(directory)
-					.redirectErrorStream(true)
-					.start();
-
-			// Here we have access to the output of the build, which we can send to GitHub.
-			String output = IOUtils.toString(process.getInputStream(), StandardCharsets.UTF_8);
-			System.err.println(output);
-
-			process.waitFor();
-		} catch (IOException e) {
-			e.printStackTrace();
-			return PipelineStatus.Fail;
-		} catch (InterruptedException e) {
-			e.printStackTrace();
-			return PipelineStatus.Fail;
-		}
-
-		return PipelineStatus.Ok;
-	}
+    /**
+     * The commands to run for the compilation
+     */
+    public final ArrayList<String> commands = new ArrayList<>();
+
+    PipelineCompiler(String... commands) {
```

Comment:
Constructor should be `public` and documented.

Reply:
Should this be changed in a different PR since it is not strictly a documentation change?

Reply:
Probably not. I guess we should open an issue where we fix the scope of all classes, functions and variables.

Reply:
Since the method is public without the public keyword, it should still be documented.

Reply:
I documented the constructor in the latest commit

Reply:
Technically it's package-scoped and not public, but in our case it's the same :)